virotour/lib/src/search.dart

```diff
@@ -0,0 +1,156 @@
+                child: const Text('OK'),
+              ),
+            ],
+          ),
+        );
+      }
+    } else {
+      throw Exception('Failed to load tour data!');
+    }
+  }
+
+  void searchData() {
+    setState(() {
+      filteredItems = [];
+      for (final item in items) {
+        String itemName = item['name'].toString().toLowerCase();
+        String itemDescription = item['description'].toString().toLowerCase();
+        String searchQueryLower = searchQuery.toLowerCase();
+        if (itemName.contains(searchQueryLower) ||
+            itemDescription.contains(searchQueryLower)) {
+          filteredItems.add(item);
+        }
+      }
+    });
+  }
+
+  void clearSearch() {
+    setState(() {
+      _searchController.clear();
+      searchQuery = '';
+      filteredItems = [];
+    });
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    fetchData();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Search'),
+      ),
+      body: Padding(
+        padding: const EdgeInsets.all(16.0),
+        child: Column(
+          children: [
+            Row(
+              children: [
+                Expanded(
+                  child: TextField(
+                    controller: _searchController,
+                    onChanged: (value) {
+                      setState(() {
+                        searchQuery = value;
+                      });
+                      searchData();
+                    },
+                    decoration: const InputDecoration(
+                      hintText: 'Enter search query',
+                    ),
+                  ),
+                ),
+                IconButton(
+                  icon: const Icon(Icons.clear),
+                  onPressed: clearSearch,
+                ),
+              ],
+            ),
+            const SizedBox(height: 16.0),
+            if (searchQuery.isNotEmpty)
+              Expanded(
+                child: ListView.builder(
+                  itemCount: filteredItems.length,
+                  itemBuilder: (BuildContext context, int index) {
+                    return Padding(
+                      padding: const EdgeInsets.symmetric(vertical: 8.0),
+                      child: GestureDetector(
+                        onTap: () {
+                          Navigator.push(
+                            context,
+                            MaterialPageRoute(
+                              builder: (context) => const TourDetailsView(),
+                            ),
+                          );
+                        },
+                        child: ElevatedButton(
+                          onPressed: () {
+                            Navigator.push(
+                              context,
+                              MaterialPageRoute(
+                                builder: (context) => const TourDetailsView(),
```

Comment:
```suggestion
                                builder: (context) => const TourDetailsView(), // Tour now requires the tour info
```
I think you'll need to update your code with the tours found from your search and pass the tour's details into the class `Tour`.
<img width="1183" alt="Screenshot 2023-03-15 at 11 12 25" src="https://user-images.githubusercontent.com/30057363/225355108-57fbed5d-bdbe-44bb-8c02-36a6c3bbc26a.png">

Reply:
Fixed.  I also needed to update the `searchData()` function.